dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/WebModuleOpenAPIProvider.java

```diff
@@ -144,4 +159,14 @@ public String getApplicationPath() {
     public boolean getServersDefined() {
         return serversDefined;
     }
+
+    @Override
+    public void waitForBackgroundTasks() {
+        try {
+            // Note: this will be called before app shutdown so we don't want to wait too long
+            asyncProcessingFuture.get(10, TimeUnit.SECONDS);
+        } catch (TimeoutException | ExecutionException | InterruptedException e) {
+            Tr.warning(tc, MessageConstants.OPENAPI_CACHE_WRITE_ERROR, webModuleInfo.getApplicationInfo().getDeploymentName(), e.toString());
```

Comment:
I think the message _"An unexpected error occurred when the OpenAPI cache was written for the {0} application. The error is"_ This works here because of the e.toString() having the notion of timeout within it but really that message only fits really well with "ExecutionException".  If e was an interrupted exception it might suggest the cache write was interrupted (which in some sense it might be if we are being deactivated on shutdown that occurs before the cache write's future completes) then we are down to the stacktrace and code knowledge to know that it was somehow the deactivation that was interrupted and not the cache write.

Reply:
It's worth noting that we don't expect either of these to happen, and if they do, there's not really a lot that the user can do about it.

For the `TimeoutException` I guess we could have a more explicit message "Timed out waiting for the OpenAPI cache to be written", though I'm not sure what we should put as the useraction here. I thought that treating it as a general unexpected exception which occurred while writing the OpenAPI cache file was reasonable.

For the `InterruptedException` I'm less sure. Thread interruption is generally a signal to give up on whatever we're doing right now. I would never expect that to occur here, except possibly if the server is shutting down? Maybe we would be better wrapping it in a `RuntimeException` and rethrowing it?

Reply:
I guess maybe for the `TimeoutException` we could instead have an informational message that we timed out waiting and are shutting the app down anyway? We could then say that no action was required.

Reply:
On server shutdown even if the background tasks is Interrupted it should complete the Future
exceptionally so we should not have to wait for an interrupted background task that got stopped
before we got deactivated. Given that, TimedoutException probably will not be seen due to waiting for a
shutdowned cache writer here (which is what I was wondering about) - so although I think perhaps what's logged could be more 'info' than 'warning' (as we are deactivating anyway) and indicates that the shutdown is perhaps not as orderly/designed as we'd expect or at least something is going on that we didn't ...it is a bit of a moot point though so I have no problem with the code staying as it is. If it is occurring unexpectedly, we will at least have something that draws our attention to it with the code as it is.